api/python/provisioner/commands/confstore_export.py

```diff
@@ -98,12 +117,23 @@ def run(self, **kwargs):
 
             Conf.load('provisioner', pillar[PillarKey(pillar_key)])
 
-            for data in template_data:
+            delimiter = '=>'
+            for i,data in enumerate(template_data):
                 if data:
-                    Conf.set("provisioner", data.split(':')[0], data.split(':')[1])
+                    key = data.split(delimiter)[0]
+                    value = data.split(delimiter)[1]
+                    if 'secret' in key:
+                        value = self._encrypt_value(key, value)
+                        template_data[i] = key + delimiter + value
+                    Conf.set("provisioner", key, value)
+
             Conf.save("provisioner")
             logger.info("Template loaded to confstore")
 
+            with open(template_file_path,'w') as stream:
+                for data in template_data:
+                    stream.writelines(data + '\n')
```

Comment:
Are you sure that this API will work here? `stream.writelines` expects an iterable of strings. So here, based on logic `data + '\n'` it looks like we are passing just a single string instead of list/tuple/any iterable object of strings. Please, double-check this logic, and if we write here only 1 string with a newline symbol consider using `stream.write()` method instead

Reply:
>Are you sure that this API will work here?
Yes. it works.
but it makes sense to use write() method. I will update it.